src/main/java/org/wildfly/security/http/HttpAuthenticator.java

```diff
@@ -98,12 +110,20 @@ public static Builder builder() {
         private volatile HttpServerMechanismsResponder successResponder;
 
         private boolean authenticate() throws HttpAuthenticationException {
+            return evaluate((ExceptionConsumer<HttpServerAuthenticationMechanism, HttpAuthenticationException>) mechanism -> mechanism.evaluateRequest(this));
+        }
+
+        private boolean logout() throws HttpAuthenticationException {
+            return evaluate((ExceptionConsumer<HttpServerAuthenticationMechanism, HttpAuthenticationException>) mechanism -> mechanism.evaluateLogoutRequest(this));
+        }
+
+        private boolean evaluate(ExceptionConsumer<HttpServerAuthenticationMechanism, HttpAuthenticationException> mechanism) throws HttpAuthenticationException {
             List<HttpServerAuthenticationMechanism> authenticationMechanisms = mechanismSupplier.get();
             responders = new ArrayList<>(authenticationMechanisms.size());
             try {
                 for (HttpServerAuthenticationMechanism nextMechanism : authenticationMechanisms) {
                     currentMechanism = nextMechanism;
-                    nextMechanism.evaluateRequest(this);
+                    mechanism.accept(nextMechanism);
 
                     if (isAuthenticated()) {
```

Comment:
I don't see anything that is going to change the outcome of this method so although logout is called isAuthenticated looks like it is going to continue to return true.